src/pages/Thanks/ThanksPage.vue

```diff
@@ -367,6 +366,20 @@ export default {
 		teamName() {
 			return this.loans?.[0]?.team?.name ?? '';
 		},
+		landedOnUSLoan() {
+			const bpPattern = /^\/lend\/(\d+)/;
+
+			if (bpPattern.test(this.$appConfig.firstPage)) {
+				const url = this.$appConfig.firstPage?.split('/');
+				const firstVisitloanId = url?.[2] ?? null;
+				const landedLoan = this.loans.find(loan => loan.id === Number(firstVisitloanId));
+				return landedLoan?.geocode?.country?.isoCode === 'US';
+			}
+			return false;
+		},
+		showNewTYPage() {
+			return !this.landedOnUSLoan && (this.isFirstLoan || this.isGuest) && !this.optedIn;
```

Comment:
How is this solution different? Was data not completed loaded on `created`?

Reply:
Because of [this if statement](https://github.com/kiva/ui/pull/5356/files#diff-c3f6821900e71ed9c13104928b17481990a08fb8bc9c8a5073a62a4adeee0b1bL498) `showNewTYPage` would only be evaluated when the user started their session on a borrower profile, preventing anyone else from seeing the new thanks page.

Reply:
Ah, we're showing this for non US loans, got it, thanks.